[CZBANK-2] fix: insufficient balance

vojtech-cerveny opened the pull request:
https://czechitas-jira.atlassian.net/browse/CZBANK-2

vojtech-cerveny committed:
fix: insufficient balance

postulkam commented on src/domain/transaction-domain/transaction-service.ts:
```diff
@@ -72,6 +72,10 @@ const transactionService = {
+      if (fromBankAccount.data.balance < amount) {
+        return errorResponse("Insufficient balance", ApiErrorCode.INSUFFICIENT_BALANCE);
+      }
+
```
Jsme si vždycky jistí, že to zde vrátí hodnotu fromBankAccount.data a nebude to `null`?

Jistější by totiž bylo použít 
`if (fromBankAccount?.data?.balance != null && fromBankAccount.data.balance < amount)`,
to nejprve ověří, že proměnná existuje (může být 0) a poté teprve bude porovnávat.

vojtech-cerveny replied:
Hmm.. jakože si nedokážu představit situaci, kdy by to mělo být `null`, ale podelat se muze vse, takze to opravim :)

vojtech-cerveny replied:
Tak null to být nemůže - Je to Float, takze tam null nebude, max 0.
```prisma
  balance         Float         @default(100000)
```

postulkam reviewed (commented):
Viz komentář u kódu.

vojtech-cerveny committed:
chore: add database setup script and update package.json

- Introduced a new script `setup-database.ts` for setting up the database, including generating the Prisma client, running migrations, and seeding the database if applicable.
- Updated `package.json` to include a new command `db:ci:setup` for executing the database setup script.

vojtech-cerveny committed:
Merge branch 'develop' into CZBANK-2-insufficient-balance

postulkam reviewed (approved):
LGTM
